types/datum.go

```diff
@@ -2248,7 +2248,7 @@ func DatumsToString(datums []Datum, handleSpecialValue bool) (string, error) {
 			return "", errors.Trace(err)
 		}
 		if datum.Kind() == KindString {
-			strs = append(strs, fmt.Sprintf("%q", str))
```

Comment:
%q will also escape the quotes. However, we can write a simple quote and escape function on our own, it should also be faster.